### PR TITLE
IS-487: Load-Testing for GGS Implementation

### PR DESCRIPTION
**Pull request by harishv7**
## PR Description

This PR aims to help load test the concurrency of git operations run in the GGS implementation.

We create X number of folders, initialise them with a sample file and perform Y number of concurrent git operations to see where is the bottleneck

There are a few main scripts:
1. `index.js`: This script creates a share simple-git instance for use across all the operations
2. `proto.js`: This script creates a new simple-git instance for each repo
3. `cleanup.js`: This script deletes the files and folder, logs etc. created while executing scripts in (1) or (2)

Note that the script takes in a `boolean` within the script (`main(false)`). The `false` indicates we do not want to push the changes to GitHub.

If you do want to test the commits and pushes to GitHub, this flag should be `true` and you need a `.env` at the folder root with `GITHUB_TOKEN` exported so that it can perform API calls to facilitate the GitHub operations.

### CODE REVIEW

File: load-test-ggs-git/index.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  // Create directory and dummy file
+  await fs.mkdir(repoDir);
+  await logInfo(`Directory ${repoDir} created.`);
+
+  const randomContent = `Random content ${Math.random()}`;
+  await fs.writeFile(`${repoDir}/test.md`, randomContent);
+  await logInfo(`test.md created in ${repoDir} with random content at time.`);
+
+  // Initialize git with cwd() method
+  await git.cwd(repoDir).init();
```
Comment (dcshzj) [inline load-test-ggs-git/index.js:51]: Sanity check here: Will the `master` branch be created here automatically? Cos from what I see there is no mention that the repo will be created with the `master` branch

Comment (harishv7) [reply load-test-ggs-git/index.js:51]: @dcshzj it will default to `main` branch so I didn't specify explicitly

**Review by alexanderleegs — APPROVED** [review]
The test mechanism looks correct to me - does this mean that we should be running a separate simple-git instance per repo since it greatly reduces errors?

The odd part about the errors in `index.js` is that the locked config file seems to be from a single repo and not the repo being written to - so this might be the cause of more concurrency errors than expected

**Comment by harishv7** [comment]
> 

Yea, somehow it is affected by another repo's .git/config which was really weird. Thinking of moving to per-repo simple-git instance for upcoming release

@dcshzj Let me know if you have any thoughts since you have researched quite deep on the simple-git stuff

**Comment by harishv7** [comment]
@alexanderleegs @dcshzj 
Summarising conversation here:

Upon testing, we found that having a shared simple-git instance for all repos works fine without errors, provided that we increase the `MAX_CONCURRENT_PROCESSES` to be equal the the number of repos we want to support concurrently.

**Pull request merged**